serdes/serde-common/src/main/java/io/apicurio/registry/serde/DefaultSchemaResolver.java

```diff
@@ -75,12 +73,10 @@ public void configure(Map<String, ?> configs, boolean isKey, SchemaParser<S> sch
 
         final ArtifactReference artifactReference = resolveArtifactReference(topic, headers, data, parsedSchema);
 
-        Long globalId = globalIdCacheByArtifactReference.get(artifactReference);
-        if (globalId != null) {
-            SchemaLookupResult<S> schema = schemaCacheByGlobalId.get(globalId);
-            if (schema != null) {
-                return schema;
-            }
+        try {
+            return resolveSchemaByArtifactReferenceCached(artifactReference);
```

Comment:
this changes drastically the previous behavior of the SchemaResolver. The idea was to check if we had that same ArtifactReference object in our cache, and use that to avoid any call to the registry. Now with this changes, this will always make calls to the registry, I don't know exactly how but IMO this needs to be changed, and if there is no way of turning back just remove this part